lib/mix/tasks/inertia.install.ex

```diff
@@ -0,0 +1,312 @@
+defmodule Mix.Tasks.Inertia.Install do
+  @moduledoc """
+  Installs and configures the Inertia.js adapter in a Phoenix application.
+
+  This installer:
+  1. Adds Inertia to the dependencies in mix.exs
+  2. Updates controller and HTML components to import Inertia functions
+  3. Adds the Inertia plug to the browser pipeline
+  4. Adds basic configuration in config.exs
+  5. Updates esbuild and configures code splitting
+  6. Set up the client-side integration packages
+
+  ## Usage
+
+      $ mix inertia.install
+
+  ## Options
+
+      --client-framework FRAMEWORK   Framework to use for the client-side integration
+                                    (react, vue, or svelte). Default is react.
+      --camelize                    Enable camelCase for props
+      --history                     Enable history support
+      --encryption                  Enable history encryption
+      --yes                         Don't prompt for confirmations
+  """
+
+  use Igniter.Mix.Task
+  require Igniter.Code.Common
+
+  @impl Igniter.Mix.Task
+  def info(_argv, _parent) do
+    %Igniter.Mix.Task.Info{
+      schema: [
+        client_framework: :string,
+        ssr: :boolean,
+        camelize: :boolean,
+        history: :boolean,
+        encryption: :boolean,
+        yes: :boolean
+      ],
+      defaults: [client_framework: "react"],
+      positional: []
+    }
+  end
+
+  @impl Igniter.Mix.Task
+  def igniter(igniter) do
+    igniter
+    |> add_inertia_dependency()
+    |> setup_controller_helpers()
+    |> setup_html_helpers()
+    |> setup_router()
+    |> add_inertia_config()
+    |> update_root_layout()
+    |> update_esbuild_config()
+    |> setup_client()
+    |> print_next_steps()
+  end
+
+  @doc false
+  def add_inertia_dependency(igniter) do
```

Comment:
This isn't necessary, as `mix igniter.install inertia` will add the dependency before running the installer.